osgar/test_logger.py

```diff
@@ -230,5 +241,52 @@ def test_large_blocks_with_time_overflow(self):
             self.assertEqual(dt, timedelta(hours=2))
         os.remove(filename)
 
+    def test_no_eof(self):
+        # similar to tail -f (--follow)
+        with LogWriter(prefix='tmpB', note='test_EOF') as log:
+            filename = log.filename
+            t1 = log.write(1, b'\x01'*100)
+            time.sleep(0.001)
+            t2 = log.write(1, b'\x02'*100)
+            time.sleep(0.001)
+            t3 = log.write(1, b'\x03'*100000)
+
+        with LogReader(filename) as log:
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x01'*100)
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x02'*100)
+
+        partial = filename + '.part'
+        with open(filename, 'rb') as f_in:
+            with open(partial, 'wb') as f_out:
+                f_out.write(f_in.read(100))
+        
+        with LogReader(partial) as log:
+            with self.assertRaises(AssertionError):
+                dt, channel, data = next(log.read_gen(only_stream_id=1))
```

Comment:
I see that you feel the need to have a `log.read` method. Why not add it? Then `log.read_gen` can be implemented using this method. This stuff with `next` seems like a hack.

Reply:
There used to be one and it is needed for tests only.

Reply:
Ok, I see now that is what you did. Thanks. I would also consider changing `read_gen` to first seek to the start of the file before returning anything so that this does not ever return.